### PR TITLE
Remote backup

### PR DESCRIPTION
**Pull request by ericm**
This would be run on the host that will be responsible for storing the credentials and triggering the backups on the proxmox hosts.
Vault vars not yet configured since I don't have the secrets yet ;)

### CODE REVIEW

File: roles/backups/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+- name: "Ensure backup cron job on each host"
+  cron:
+    user: "root"
+    name: "Run VZdump and duplicity"
+    hour: "{{ backup_hour }}"
+    minute: "0"
+    job: >
+      bash -c "
+      scp -r -i /root/NaC/keys/{{ item }}/id_rsa /root/.duplicity netsoc@{{ item }}:/root/.duplicity &&
+      ssh -i /root/NaC/keys/{{ item }}/id_rsa netsoc@{{ item }} 'bash -s' < /root/backup.sh"
```
Comment (ocanty) [inline roles/backups/tasks/main.yml:105]: don't hardcode the keypath, let you specify them as a role var

File: setup-backups.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    # Get these by generating .duplicity_onedrive_oauthtoken.json with duplicity on your own machine.
+    onedrive_access_token: "{{ vault_onedrive_access_token }}"
+    onedrive_refresh_token: "{{ vault_onedrive_refresh_token }}"
+    onedrive_user_id: "{{ vault_onedrive_user_id: }}"
+
+    onedrive_location: "{{ vault_onedrive_location }}"
+
+    backup_sign_key: "{{ vault_backup_sign_key }}"
+    backup_sign_key_passphrase: "{{ vault_backup_sign_key_passphrase }}"
+    backup_encryptionn_key:_key: "{{ vault_backup_encryptionn_key:_key }}"
```
Comment (ocanty) [inline setup-backups.yml:18]: ``nn``

File: roles/backups/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      for VM in $(qm list --full | 
+      awk '/vm.netsoc.co/ { if ($2 != "{{ blacklist }}") { printf "%s\n\n", $1; } }') ;
+        do CONF=`pvesh get /nodes/$(hostname -s)/qemu/$VM/config --output-format=json` &&
+          [[ `echo $CONF | jq .virtio0` != "null" ]] && [[ `echo $CONF | jq .virtio0` != *"backup=0"* ]] && pvesh set /nodes/$(hostname -s)/qemu/$VM/config -virtio0 `echo $CONF | jq -r .virtio0`,backup=0 && sleep 5;
+          [[ `echo $CONF | jq .virtio1` != "null" ]] && 
+          echo "Backing up $VM" && MDIR="/backup/$(echo $CONF | jq -r .name)" && 
+          mkdir -p $MDIR && vzdump --compress {{ compress }} $VM --mode snapshot --dumpdir $MDIR --exclude-path /boot ||
+            curl -H "Content-Type: application/json" -X POST -d '{"username": "$(hostname -s)", "content": "Error backing up VM $VM"}' {{ discord_webhook_url }};
+        done;
+        duplicity-backup.sh --config /root/.dupliticy/duplicity-backup.conf --full ||
```
Comment (ocanty) [inline roles/backups/tasks/main.yml:89]: ` /root/.dupliticy/`

**Comment by ericm** [comment]
@ocanty backups should be ready to deploy now. will write a wiki entry if you're happy.

**Comment by ocanty** [comment]
lgtm, great work

File: setup-proxmox-host-initial.yml [hunk 4]
```diff
@@ -5,6 +5,7 @@
   roles:
     - ansible
     - ifupdown2
+    - roles/backups/tasks/dependencies.yml
```
Comment (ocanty) [inline setup-proxmox-host-initial.yml:8]: this could break NaC

**Pull request merged**